# Знакомство с данными

Три файла задания: `train.parquet` — пары «запрос → выбранное объявление», `benchmark_queries.parquet` — запросы без разметки, `benchmark_items.parquet` — корпус, из которого нужно выбирать кандидатов.

Хочу понять, что считать запросом, насколько train похож на бенчмарк и какие поля пригодятся для поиска.

In [ ]:
from pathlib import Path

import pandas as pd

DATA_DIR = Path("../dataset")

In [ ]:
train = pd.read_parquet(DATA_DIR / "train.parquet")
queries = pd.read_parquet(DATA_DIR / "benchmark_queries.parquet")
items = pd.read_parquet(DATA_DIR / "benchmark_items.parquet")

print("Размеры:")
print("train:", train.shape)
print("queries:", queries.shape)
print("items:", items.shape)

Размеры:
train: (497673, 19)
queries: (2452, 6)
items: (189212, 14)


In [ ]:
train.head()

,search_query,search_location_id,search_is_delivery_search,search_infm_params_text,search_category,item_title_raw,item_rating_reviews_count,item_rating,item_price,item_microcat_id,item_longitude,item_location_id,item_latitude,item_is_phone_hidden,item_is_message_forbidden,item_infm_params_text,item_id,item_description_raw,item_category_id
0,скупка телевизоров,652430,0,,114,Скупка б/у техники,91.0,4.989011,1.000000000000000,2303374,39.712818145751953,652430,54.629230499267578,True,False,Вид услуги Место оказания услуг Первомайский п...,e8b685dffe1a408e,Скупаю практически любую современную новую и б...,114
1,автоподбор,640860,0,Рейтинг пользователя 4 звезды и выше,114,Автоподбор Разовый осмотр автомобиля,462.0,4.982684,3500.000000000000000,2303374,44.051009999999998,640860,56.273389999999999,False,False,Вид услуги Место оказания услуг Нижний Новгоро...,92e1b0446f827b59,🚗 Автоподбор и выездная диагностика автомобиля...,114
2,баня на дровах,653240,0,"Онлайн-запись Тип услуги СПА-услуги, массаж Ви...",114,"Баня на дровах ""Прованс"" на Цветочной",NaN,NaN,1600.000000000000000,86469,30.128784180000000,653240,59.783687590000000,False,False,"Вид услуги Красота, здоровье Место оказания ус...",624846856ce81d69,В ритме современной жизни так сложно найти мо...,114
3,изготовление госномера на авто,634670,0,"Вид услуги Оборудование, производство",114,"Изготовление дубликатов авто номеров, гос номеров",14.0,4.714286,1700.000000000000000,2303428,40.537841000000000,633570,45.424875000000000,False,False,"Вид услуги Оборудование, производство Тип услу...",45b8628b9c6e7b85,Изготовим дубликат номера по утере или износу ...,114
4,укладка плитки,658430,0,Тип услуги Ремонт квартир и домов под ключ Вид...,114,Ремонт и отделка квартир под ключ,1.0,5.000000,1000.000000000000000,44725,69.496444699999998,658430,56.105983729999998,False,False,Вид услуги Ремонт и отделка Тип услуги Ремонт ...,c95a4a7daf2a967f,отделочные работы любой сложности.,114


In [ ]:
queries.head()

,query_id,search_query,search_location_id,search_is_delivery_search,search_infm_params_text,search_category
0,70DfDUpwjxB4lzFd,перевозки владикавказ тбилиси,649820,0,,114
1,JTrdTaZJvSiLPkXj,обзвон по базе,107620,0,Вид услуги Деловые услуги,114
2,LZCZNoVG4AFUkVRJ,липоредукция подбородка,637640,0,"Вид услуги Красота, здоровье",114
3,660ac9QVtXkRxZC3,подъемник 4 х стоечный,662810,0,,114
4,YgHcM9MVbxKnxD1e,монтаж видеодомофонов,642790,0,,114


In [ ]:
items.head()

,item_title_raw,item_rating_reviews_count,item_rating,item_price,item_microcat_id,item_longitude,item_location_id,item_latitude,item_is_phone_hidden,item_is_message_forbidden,item_infm_params_text,item_id,item_description_raw,item_category_id
0,Ремонт/выкуп компьют. и ноутбуков с выездом на...,57.0,5.0,500.000000000000000,2097573,42.047193961004901,631060,44.228180450924000,False,False,Вид услуги Компьютерная помощь Место оказания ...,111eb8b979577d79,"Выезд на дом в любое время, вплоть до 23:00\n\...",114
1,Обучение ребенка чтению,7.0,5.0,900.000000000000000,86453,49.627391820000000,631870,58.627609249999999,False,False,"Вид услуги Обучение, курсы Место оказания услу...",75fc8e10f5a66fc4,"Дорогие родители и маленькие книголюбы! \n\nЯ,...",114
2,Афрокудри 5+,14.0,5.0,1000.000000000000000,86467,40.936892000000000,628500,56.990307000000001,True,False,"Вид услуги Красота, здоровье Место оказания ус...",3f6ae81704565b9c,ВНИМАНИЕ Укладка !!!! 🥳\n\nАФРОкудри отличный ...,114
3,Монтаж малых архитектурных форм,2.0,5.0,5000.000000000000000,2058739,37.576183000000000,637640,55.662734999999998,False,False,Вид услуги Строительство Место оказания услуг ...,0618dec37a59a21a,Оказываем все виды услуг по сборке и монтажу и...,114
4,Репетитор по истории 10 класс ЕГЭ,19.0,5.0,1500.000000000000000,86456,44.750473999999997,624850,48.786008000000002,False,False,"Вид услуги Обучение, курсы Место оказания услу...",13da2c81574677ed,🚀 Сдай ЕГЭ по истории на 80+ за 6 месяцев!\n\n...,114


Пропусков почти нет: рейтинг и число отзывов есть не у всех объявлений, у четырёх строк train нет координат.

In [ ]:
print(train.isna().sum())
print(queries.isna().sum())
print(items.isna().sum())

search_query                     0
search_location_id               0
search_is_delivery_search        0
search_infm_params_text          0
search_category                  0
item_title_raw                   0
item_rating_reviews_count    19211
item_rating                  28232
item_price                       0
item_microcat_id                 0
item_longitude                   4
item_location_id                 0
item_latitude                    4
item_is_phone_hidden             0
item_is_message_forbidden        0
item_infm_params_text            0
item_id                          0
item_description_raw             0
item_category_id                 0
dtype: int64
query_id                     0
search_query                 0
search_location_id           0
search_is_delivery_search    0
search_infm_params_text      0
search_category              0
dtype: int64
item_title_raw                   0
item_rating_reviews_count    11615
item_rating                  17631
item_price        

In [ ]:
print(train['item_price'].apply(type).value_counts())
print(train['item_price'].head(10).tolist())

print(train['item_latitude'].apply(type).value_counts())
print(train['item_latitude'].head(10).tolist())

item_price
<class 'decimal.Decimal'>    497673
Name: count, dtype: int64
[Decimal('1.000000000000000'), Decimal('3500.000000000000000'), Decimal('1600.000000000000000'), Decimal('1700.000000000000000'), Decimal('1000.000000000000000'), Decimal('1500.000000000000000'), Decimal('700.000000000000000'), Decimal('4000.000000000000000'), Decimal('100000.000000000000000'), Decimal('1000.000000000000000')]
item_latitude
<class 'decimal.Decimal'>    497669
<class 'NoneType'>                4
Name: count, dtype: int64
[Decimal('54.629230499267578'), Decimal('56.273389999999999'), Decimal('59.783687590000000'), Decimal('45.424875000000000'), Decimal('56.105983729999998'), Decimal('55.746436000000003'), Decimal('55.742385859999999'), Decimal('55.918952939999997'), Decimal('55.174182999999999'), Decimal('48.490543369999997')]


## Пересечение train и корпуса бенчмарка

Если бы объявления бенчмарка встречались в train, можно было бы опираться на историю кликов по конкретным `item_id`.

In [ ]:
train_item_ids = set(train["item_id"].dropna().unique())
corpus_item_ids = set(items["item_id"].dropna().unique())

common_item_ids = train_item_ids & corpus_item_ids

print("Уникальных item_id в train:", len(train_item_ids))
print("Уникальных item_id в items:", len(corpus_item_ids))
print("Пересечение:", len(common_item_ids))

Уникальных item_id в train: 344825
Уникальных item_id в items: 189212
Пересечение: 18142


Общих объявлений 18 142 из 189 212, около 10%. Искать нужно по содержанию — тексту, категории, локации.

## Уникальные

74.5 тыс. уникальных текстов


In [ ]:
print("Уникальных search_query (по тексту):", train['search_query'].nunique())
print("Строк train:", len(train))
print("Среднее число объявлений на уникальный текст запроса:",
      len(train) / train['search_query'].nunique())

print(train.groupby('search_query').size().describe())

Уникальных search_query (по тексту): 74529
Строк train: 497673
Среднее число объявлений на уникальный текст запроса: 6.677575172080666
count    74529.000000
mean         6.677575
std         64.865220
min          1.000000
25%          1.000000
50%          1.000000
75%          3.000000
max       6540.000000
dtype: float64


In [ ]:
print(train['search_query'].value_counts().head(15))

search_query
маникюр                    6540
массаж                     5702
наращивание ресниц         5395
установка кондиционеров    4573
педикюр                    3436
сантехник                  3405
вывоз мусора               3278
эвакуатор                  3263
электрик                   3158
автоэлектрик               2790
покос травы                2477
вспашка земли              2315
манипулятор                2120
грузоперевозки газель      2034
натяжные потолки           2004
Name: count, dtype: int64


## Дубликаты

Сначала дубли пар «текст запроса — объявление».

In [ ]:
print("Дублирующихся пар (search_query, item_id):",
      train.duplicated(subset=['search_query', 'item_id']).sum())

Дублирующихся пар (search_query, item_id): 52412


Часть из них — один и тот же текст в разных городах или с разными фильтрами, то есть разные запросы. Настоящий дубль — совпадение всех полей `search_*` и `item_id`.

In [ ]:
search_cols = ['search_query', 'search_location_id', 'search_is_delivery_search',
               'search_infm_params_text', 'search_category']

print("Уникальных комбинаций search_* полей:", train[search_cols].drop_duplicates().shape[0])

dup_mask = train.duplicated(subset=['search_query', 'item_id'], keep=False)
dup_rows = train[dup_mask].sort_values('search_query')

check = dup_rows.groupby(['search_query', 'item_id'])[search_cols[1:]].nunique()
print(check.describe())
print(check[(check > 1).any(axis=1)].head(10))

full_dup = train.duplicated(subset=search_cols + ['item_id'], keep=False)
print("Дублей по ВСЕМ search_* полям + item_id:", full_dup.sum())

Уникальных комбинаций search_* полей: 354463
       search_location_id  search_is_delivery_search  search_infm_params_text  \
count        32006.000000                    32006.0             32006.000000   
mean             1.245641                        1.0                 1.536493   
std              0.454345                        0.0                 0.574054   
min              1.000000                        1.0                 1.000000   
25%              1.000000                        1.0                 1.000000   
50%              1.000000                        1.0                 2.000000   
75%              1.000000                        1.0                 2.000000   
max              6.000000                        1.0                 6.000000   

       search_category  
count     32006.000000  
mean          1.000031  
std           0.005590  
min           1.000000  
25%           1.000000  
50%           1.000000  
75%           1.000000  
max           2.000000  


Полные дубли не идут подряд и стоят на разных местах в файле — похоже на одинаковые клики разных пользователей, а не на технический артефакт. Для валидации их можно удалить.

In [ ]:
search_cols = ['search_query', 'search_location_id', 'search_is_delivery_search',
               'search_infm_params_text', 'search_category']
full_key = search_cols + ['item_id']

full_dup_mask = train.duplicated(subset=full_key, keep=False)
full_dups = train[full_dup_mask].sort_values(full_key)

print("Всего строк в полных дублях (все вхождения):", len(full_dups))
print("Из них уникальных комбинаций (сколько 'групп' дублей):",
      full_dups.drop_duplicates(subset=full_key).shape[0])

dup_counts = full_dups.groupby(full_key).size().sort_values(ascending=False)
print(dup_counts.describe())
print(dup_counts.head(10))

top_dup_key = dup_counts.index[0]
example = train[
    (train['search_query'] == top_dup_key[0]) &
    (train['search_location_id'] == top_dup_key[1]) &
    (train['search_is_delivery_search'] == top_dup_key[2]) &
    (train['search_infm_params_text'] == top_dup_key[3]) &
    (train['search_category'] == top_dup_key[4]) &
    (train['item_id'] == top_dup_key[5])
]
print(example)

pairs_count2 = dup_counts[dup_counts == 2].index[:5]
for key in pairs_count2:
    subset = train[
        (train['search_query'] == key[0]) &
        (train['search_location_id'] == key[1]) &
        (train['search_is_delivery_search'] == key[2]) &
        (train['search_infm_params_text'] == key[3]) &
        (train['search_category'] == key[4]) &
        (train['item_id'] == key[5])
    ]
    print(subset[['search_query', 'item_id'] + [c for c in subset.columns if c not in full_key]])
    print('---')

Всего строк в полных дублях (все вхождения): 48533
Из них уникальных комбинаций (сколько 'групп' дублей): 17914
count    17914.000000
mean         2.709222
std          3.687829
min          2.000000
25%          2.000000
50%          2.000000
75%          2.000000
max        171.000000
dtype: float64
search_query                                    search_location_id  search_is_delivery_search  search_infm_params_text                                   search_category  item_id         
увеличение губ                                  653240              0                          Тип услуги Косметология Вид услуги Красота, здоровье      114              391037f10a20ee24    171
остекление лоджии и балконов - ремонт под ключ  653240              0                          Тип услуги Окна и балконы Вид услуги Ремонт и отделка     114              4be48741454a11dc    126
остекление лоджий по ключ                       653240              0                          Тип услуги Окна и балконы В

       search_query           item_id                  item_title_raw  \
160440    видеограф  c28a340b403d9a0e  Свадебный видеограф Красноярск   
457363    видеограф  c28a340b403d9a0e  Свадебный видеограф Красноярск   

        item_rating_reviews_count  item_rating            item_price  \
160440                        2.0          5.0  1000.000000000000000   
457363                        2.0          5.0  1000.000000000000000   

        item_microcat_id      item_longitude  item_location_id  \
160440           2301875  92.868409000000000            635320   
457363           2301875  92.868409000000000            635320   

             item_latitude  item_is_phone_hidden  item_is_message_forbidden  \
160440  56.008645999999999                 False                      False   
457363  56.008645999999999                 False                      False   

                                    item_infm_params_text  \
160440  Вид услуги Фото- и видеосъёмка Место оказания ...   
4573

Одно объявление может выбираться по очень многим запросам — вот самое популярное:

In [ ]:
train[train["item_id"]=="391037f10a20ee24"]

,search_query,search_location_id,search_is_delivery_search,search_infm_params_text,search_category,item_title_raw,item_rating_reviews_count,item_rating,item_price,item_microcat_id,item_longitude,item_location_id,item_latitude,item_is_phone_hidden,item_is_message_forbidden,item_infm_params_text,item_id,item_description_raw,item_category_id
4930,увеличение губ,653240,0,"Тип услуги Косметология Вид услуги Красота, зд...",114,"Увеличение губ,Фулл Фэйс,м.Комендантский и Озерки",123.0,4.934959,6990.000000000000000,2096640,30.318330992856225,653240,60.041219693112787,False,False,"Вид услуги Красота, здоровье Место оказания ус...",391037f10a20ee24,Описание\n❗️6900 за 1 мл❗️Цeнa без дoплат‼️Oпы...,114
5533,увеличение губ,653240,0,"Тип услуги Косметология Вид услуги Красота, зд...",114,"Увеличение губ,Фулл Фэйс,м.Комендантский и Озерки",123.0,4.934959,6990.000000000000000,2096640,30.318330992856225,653240,60.041219693112787,False,False,"Вид услуги Красота, здоровье Место оказания ус...",391037f10a20ee24,Описание\n❗️6900 за 1 мл❗️Цeнa без дoплат‼️Oпы...,114
5996,увеличение губ,653240,0,"Тип услуги Косметология Вид услуги Красота, зд...",114,"Увеличение губ,Фулл Фэйс,м.Комендантский и Озерки",123.0,4.934959,6990.000000000000000,2096640,30.318330992856225,653240,60.041219693112787,False,False,"Вид услуги Красота, здоровье Место оказания ус...",391037f10a20ee24,Описание\n❗️6900 за 1 мл❗️Цeнa без дoплат‼️Oпы...,114
9407,увеличение губ,653240,0,"Тип услуги Косметология Вид услуги Красота, зд...",114,"Увеличение губ,Фулл Фэйс,м.Комендантский и Озерки",123.0,4.934959,6990.000000000000000,2096640,30.318330992856225,653240,60.041219693112787,False,False,"Вид услуги Красота, здоровье Место оказания ус...",391037f10a20ee24,Описание\n❗️6900 за 1 мл❗️Цeнa без дoплат‼️Oпы...,114
13660,увеличение губ,653240,0,"Тип услуги Косметология Вид услуги Красота, зд...",114,"Увеличение губ,Фулл Фэйс,м.Комендантский и Озерки",123.0,4.934959,6990.000000000000000,2096640,30.318330992856225,653240,60.041219693112787,False,False,"Вид услуги Красота, здоровье Место оказания ус...",391037f10a20ee24,Описание\n❗️6900 за 1 мл❗️Цeнa без дoплат‼️Oпы...,114
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
480687,увеличение губ,653240,0,"Тип услуги Косметология Вид услуги Красота, зд...",114,"Увеличение губ,Фулл Фэйс,м.Комендантский и Озерки",123.0,4.934959,6990.000000000000000,2096640,30.318330992856225,653240,60.041219693112787,False,False,"Вид услуги Красота, здоровье Место оказания ус...",391037f10a20ee24,Описание\n❗️6900 за 1 мл❗️Цeнa без дoплат‼️Oпы...,114
482630,увеличение губ,653240,0,"Тип услуги Косметология Вид услуги Красота, зд...",114,"Увеличение губ,Фулл Фэйс,м.Комендантский и Озерки",123.0,4.934959,6990.000000000000000,2096640,30.318330992856225,653240,60.041219693112787,False,False,"Вид услуги Красота, здоровье Место оказания ус...",391037f10a20ee24,Описание\n❗️6900 за 1 мл❗️Цeнa без дoплат‼️Oпы...,114
491646,увеличение губ,653240,0,"Тип услуги Косметология Вид услуги Красота, зд...",114,"Увеличение губ,Фулл Фэйс,м.Комендантский и Озерки",123.0,4.934959,6990.000000000000000,2096640,30.318330992856225,653240,60.041219693112787,False,False,"Вид услуги Красота, здоровье Место оказания ус...",391037f10a20ee24,Описание\n❗️6900 за 1 мл❗️Цeнa без дoплат‼️Oпы...,114
493492,увеличение губ,653240,0,"Тип услуги Косметология Вид услуги Красота, зд...",114,"Увеличение губ,Фулл Фэйс,м.Комендантский и Озерки",123.0,4.934959,6990.000000000000000,2096640,30.318330992856225,653240,60.041219693112787,False,False,"Вид услуги Красота, здоровье Место оказания ус...",391037f10a20ee24,Описание\n❗️6900 за 1 мл❗️Цeнa без дoплат‼️Oпы...,114


## Категория

В бенчмарке `search_category` почти всегда 114, у 222 запросов — 0. Как сигнал для выбора объявлений категория почти ничего не даёт

In [ ]:
queries["search_category"].value_counts()

114    2230
0       222
Name: search_category, dtype: int64

## Фильтры запроса

`search_infm_params_text` — выбранные пользователем фильтры, например «Вид услуги Ремонт и отделка». Бывают запросы, где заполнено только название фильтра без значения.

In [ ]:
queries[queries["search_infm_params_text"]=="Вид услуги"]

,query_id,search_query,search_location_id,search_is_delivery_search,search_infm_params_text,search_category
111,NjWdw4HSKj2LmpYH,экскурсии лазаревское,633655,0,Вид услуги,114
117,jhqvJdOYB0SP0KqJ,скупка фотоаппаратов,107621,0,Вид услуги,114
215,SVFfhQTJ5W0IOJtK,теплая беседка,634930,0,Вид услуги,114
298,Rv9yM596h15myHBU,авто разбор,645530,0,Вид услуги,114
484,8pmTLkEsss3S0BUq,надпись на футболке печать,637640,0,Вид услуги,114
540,VLa9UWTGQaGOXQYR,от шершней,646600,0,Вид услуги,114
577,hXCl0sxTmJFpVz0u,футболки с принтами на заказ,637640,0,Вид услуги,114
609,0mtMahErLgwnbCRg,отравление тараканов,661950,0,Вид услуги,114
628,3Ck0jwB6hlydPX2V,студия для фотосессии,637640,0,Вид услуги,114
648,no5QcEFwHNu79q7v,стоматолог,642320,0,Вид услуги,114


In [ ]:
train[train["search_infm_params_text"]=="Вид услуги"]

,search_query,search_location_id,search_is_delivery_search,search_infm_params_text,search_category,item_title_raw,item_rating_reviews_count,item_rating,item_price,item_microcat_id,item_longitude,item_location_id,item_latitude,item_is_phone_hidden,item_is_message_forbidden,item_infm_params_text,item_id,item_description_raw,item_category_id
8,выкуп компьютеров,661420,0,Вид услуги,114,"Выкуп пк, ноутбуков,приставок",1071.0,4.984127,100000.000000000000000,2303374,61.308090000000000,661420,55.174182999999999,False,False,Вид услуги Место оказания услуг ул. Братьев Ка...,1806b31ef88a665f,"Срoчный Выкуп Bидeокарт,Ноутбуков,Mайнинг фeрм...",114
27,выкуп авто,650520,0,Вид услуги,114,Выкуп авто,6.0,5.000000,-1.000000000000000,2303374,51.809113000000004,650520,55.634070999999999,False,False,Вид услуги Место оказания услуг пр-т Химиков Т...,c74b4bb618a08ecc,7/24 срочно выкуп авто в любом состоянии! биты...,114
28,неоновая вывеска,633080,0,Вид услуги,114,Неоновая вывеска. Хит продаж,115.0,4.913043,650.000000000000000,2302479,38.077114999999999,633080,44.561011999999998,True,False,Вид услуги Место оказания услуг Краснодарский ...,5bb11051b1e108d0,⚡ Неоновые вывески от NEON PREMIUM!\n\n🧑‍💻 Нап...,114
150,изготовление печатей,623760,0,Вид услуги,114,Печати и штампы без документов,183.0,4.994536,300.000000000000000,2302479,37.833230999999998,623760,51.297974000000004,True,False,Вид услуги Место оказания услуг Белгородская о...,9828d1720d7e1b79,"Изготавливаем Печати и штампы по оттиску , от ...",114
253,выкуп авто,656901,0,Вид услуги,114,Выкуп авто сегодня - Без торга,56.0,5.000000,55555555.000000000000000,2303374,35.917474644180274,656901,56.857100231431772,False,False,"Вид услуги Место оказания услуг Тверь, улица А...",b0805d4de1a3e432,"🚗 Хотите продать автомобиль быстро, без размещ...",114
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
497423,печать баннеров,654070,0,Вид услуги,114,"Екатеринбург. Печать,монтаж рекламы;баннеры,пл...",19.0,5.000000,-1.000000000000000,2302479,60.668136599999997,654070,56.746608729999998,False,False,Вид услуги Место оказания услуг Свердловская о...,cc1a2f72c0a5b8f6,Мы рекламное агентство Image Media работаем с ...,114
497449,срочный выкуп авто,621590,0,Вид услуги,114,Выкуп автомобилей в любом состояние,103.0,4.815534,-1.000000000000000,2303374,83.774383540000002,621630,53.364456179999998,False,False,Вид услуги Место оказания услуг Алтайский край...,2e351d728c5b106a,Срочный выкуп автомобилей\nНужна быстрая прода...,114
497489,тараканы,631870,0,Вид услуги,114,"Дезинсекция тараканов, уничтожение клопов, блох",79.0,4.962025,330.000000000000000,2301738,49.644658000000000,631870,58.588833000000001,False,False,Вид услуги Место оказания услуг Кировская обл....,46249342e2345667,🪳 Дезинсекция от насекомых и грызунов — профес...,114
497499,копицентр,625810,0,Вид услуги,114,"Наружная реклама, вывески, объемные буквы, неон",66.0,5.000000,555.000000000000000,2302479,39.200296000000002,625810,51.660781000000000,False,False,Вид услуги Место оказания услуг Воронежская об...,7bbcffdfe896eceb,"СДЕЛАЕМ ВЫВЕСКУ ЯРЧЕ И КРАСИВЕЕ, ЧЕМ У ВАШИХ К...",114


## Локация

Один и тот же текст ищут в сотнях городов: «маникюр» встречается в 633 локациях.

In [ ]:
train[train["search_query"]=="маникюр"]["item_location_id"].value_counts()

item_location_id
637640    507
653240    501
654070    130
650400    127
641780    121
         ... 
625720      1
629210      1
656930      1
656820      1
659940      1
Name: count, Length: 633, dtype: int64

Внутри одной локации выбор тоже неравномерный: несколько объявлений собирают заметную долю кликов.

In [ ]:
train[train["item_location_id"] == 652430]["item_id"].value_counts()

dc090a318115e720    18
142f24a97b5713a9    17
a7ea0a63cbc5be25    12
719f4fb8c18882f5    12
a90e51853233190d    12
                    ..
b453ca184ad48f89     1
0b9e9ff7b812dcb2     1
f2e7cc87c0dcb62c     1
82b25cc859d495f5     1
9759974b0c26af4d     1
Name: item_id, Length: 2248, dtype: int64

История есть не для всех запросов: например, по «нарколог» в train нет ни одной строки.

In [ ]:
train[train["search_query"]=="нарколог"]

,search_query,search_location_id,search_is_delivery_search,search_infm_params_text,search_category,item_title_raw,item_rating_reviews_count,item_rating,item_price,item_microcat_id,item_longitude,item_location_id,item_latitude,item_is_phone_hidden,item_is_message_forbidden,item_infm_params_text,item_id,item_description_raw,item_category_id


## Выводы

- Запрос — полная комбинация полей `search_*`: один текст в разных локациях и с разными фильтрами даёт разные наборы объявлений.
- Объявления train и бенчмарка почти не пересекаются, поэтому поиск должен работать по содержанию.
- Полные дубли строк — повторные клики, перед валидацией их удаляю.
- Цена и координаты требуют приведения типов.
- Локация выглядит сильным сигналом, её вклад проверяю в `02_experiments`.